packages/IntegrationTest/windows/integrationtest/TestHostHarness.cpp

```diff
@@ -187,27 +187,73 @@ void TestHostHarness::CompletePendingResponse() noexcept {
   }
 }
 
-void TestHostHarnessRedboxHandler::ShowNewError(IRedBoxErrorInfo info, RedBoxErrorType /*type*/) {
+void TestHostHarnessRedboxHandler::ShowNewError(const IRedBoxErrorInfo &info, RedBoxErrorType /*type*/) noexcept {
+  m_pendingException = std::make_unique<ExceptionInfo>();
```

Comment:
>m_pendingException [](start = 2, length = 18)

Question: Do we risk race conditions with multiple errors firing from various threads and them trampling over eachother?
If not, perhaps an extra line to the comment at the end of the function.

Reply:
Can check later, but I think the RedboxHandler is always called on the same thread. I.e. we shouldn't have possible races for updating exception info, but it's a different thread than the test host, so we need to be careful about how we handle errors once we deliver the notification to the UI thread. Right now the strategy is basically to fail the test with the first error the UI thread sees, then flush any pending messages before starting another test and creating a new pending response.